src/backend/aspen/app/views/phylo_runs.py

```diff
@@ -0,0 +1,188 @@
+    )
+
+
+class GroupResponseSchema(Schema):
+    id = fields.Int()
+    name = fields.String()
+    address = fields.String()
+    prefix = fields.String()
+    division = fields.String()
+    location = fields.String()
+
+
+class WorkflowStatusSchema(Schema):
+    name = fields.String()
+
+
+class PhyloRunResponseSchema(Schema):
+    id = fields.Int()
+    start_datetime = fields.DateTime()
+    end_datetime = fields.DateTime()
+    workflow_status = fields.Pluck(WorkflowStatusSchema, "name")
+    group = fields.Nested(GroupResponseSchema, only=("id", "name"))
+    template_file_path = fields.String()
+    template_args = fields.Nested(GroupResponseSchema, only=("division", "location"))
+
+
+@application.route("/api/phylo_runs", methods=["POST"])
+@requires_auth
+def start_phylo_run():
+    user = g.auth_user
+    # Note - sample run will be associated with users's primary group.
+    #    (do we want admins to be able to start runs on behalf of other dph's ?)
+    group = user.group
+
+    # Step 1 - Validate our request body
+    validator = PhyloRunRequestSchema()
+    try:
+        request_data = validator.load(request.get_json())
+    except ValidationError as verr:
+        return Response(str(verr), 400)
+    sample_ids = request_data["samples"]
+
+    # Step 2 - prepare big sample query per the old db cli
+    all_samples: Iterable[Sample] = g.db_session.query(Sample).options(
+        joinedload(Sample.uploaded_pathogen_genome, innerjoin=True),
+    )
+    # Step 3 - Enforce AuthZ
+    all_samples = authz_sample_filters(all_samples, sample_ids, user)
+
+    pathogen_genomes: MutableSequence[PathogenGenome] = list()
+    missing_sample_ids = list(sample_ids)  # Make a copy of sample id's
+    for sample in all_samples:
+        pathogen_genomes.append(sample.uploaded_pathogen_genome)
+        try:
+            missing_sample_ids.remove(sample.private_identifier)
+        except ValueError:
+            pass
+        try:
+            missing_sample_ids.remove(sample.public_identifier)
+        except ValueError:
+            pass
+    if len(pathogen_genomes) == 0:
+        raise ValueError("No sequences selected for run.")
+
+    if missing_sample_ids:
+        sentry_sdk.capture_message(
+            f"User requested sample id's they didn't have access to ({missing_sample_ids})"
+        )
+
+    aligned_gisaid_dump = (
+        g.db_session.query(AlignedGisaidDump)
+        .join(AlignedGisaidDump.producing_workflow)
+        .order_by(Workflow.end_datetime.desc())
+        .first()
+    )
+    if not aligned_gisaid_dump:
+        return Response("No gisaid dump for run", 500)
+
+    template_path_prefix = (
+        "/usr/src/app/aspen/workflows/nextstrain_run/builds_templates"
+    )
+    builds_template_file = (
+        f"{template_path_prefix}/{PHYLO_TREE_TYPES[request_data['tree_type']]}"
+    )
+    builds_template_args = {
+        "division": group.division,
+        "location": group.location,
```

Comment:
i think this should be the sample division/ location and not the group division /location?

Reply:
Yeah - this is a little sticky actually. Theoretically each sample can have a different division/location, but the value in this part of the code is what get's populated in the nextstrain config for filtering, and we can only specify a single division/location there, which is why I was leaning towards using the group's division/location. Thoughts? @danrlu ?

Reply:
Very good question. I agree with what Jessica said!